src/GoogleCalendarFactory.php

```diff
@@ -19,14 +19,43 @@ public static function createForCalendarId(string $calendarId): GoogleCalendar
     }
 
     public static function createAuthenticatedGoogleClient(array $config): Google_Client
+    {
+        $authProfile = $config['default_auth_profile'];
+
+        switch ($authProfile) {
+            case 'service_account':
+                return self::createServiceAccountClient($config['auth_profiles']['service_account']);
+            case 'oauth':
+                return self::createOAuthClient($config['auth_profiles']['oauth']);
+        }
+
+        throw new \InvalidArgumentException("Unsupported authentication profile [{$authProfile}].");
+    }
+
+    protected static function createServiceAccountClient(array $authProfile): Google_Client
     {
         $client = new Google_Client;
 
         $client->setScopes([
             Google_Service_Calendar::CALENDAR,
         ]);
 
-        $client->setAuthConfig($config['service_account_credentials_json']);
+        $client->setAuthConfig($authProfile['credentials_json']);
```

Comment:
If these calls are chainable, please chain them.

Reply:
Calls are not chainable